sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/processing/flow/FlowThen.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2022 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.ui.labmessage.processing.flow;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Class used for initializing and building async flow/chain
+ * 
+ * @param <R>
+ */
+public class FlowThen<R> {
+
+	private final CompletionStage<ProcessingResult<R>> currentResult;
+
+	public FlowThen() {
+		this(ProcessingResult.<R> continueWith(null).asCompletedFuture());
+	}
+
+	public FlowThen(CompletionStage<ProcessingResult<R>> currentResult) {
+		this.currentResult = currentResult;
+	}
+
+	public <RR> FlowThen<RR> then(FlowAction<R, RR> action) {
```

Comment:
Also here:

`SonarLint: Rename this generic name to match the regular expression '^[A-Z][0-9]?$'.`